pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -142,8 +138,15 @@ func (c *Controller) List(typ config.GroupVersionKind, namespace string) ([]conf
 	}
 }
 
-func (c *Controller) SetStatusWrite(enabled bool) {
+func (c *Controller) SetStatusWrite(enabled bool, statusManager *status.Manager) {
 	c.statusEnabled.Store(enabled)
+	if enabled && features.EnableGatewayAPIStatus && statusManager != nil {
+		c.statusController = statusManager.CreateGenericController(func(status interface{}, context interface{}) status.GenerationProvider {
```

Comment:
is this race-y? I think we don't have a lock here?

Reply:
which of these checks would be racy?